src/Core/Domain/Order/QueryResult/OrderForViewing.php

```diff
@@ -396,6 +396,14 @@ public function getPayments(): OrderPaymentsForViewing
         return $this->payments;
     }
 
+    /**
+     * @return bool
+     */
+    public function hasPayments(): bool
+    {
+        return count($this->payments->getPayments()) > 0;
```

Comment:
```suggestion
        return count($this->payments->getPayments());
```

Reply:
Even though a count can't be negative and the code you propose will also work I prefer to be explicit Besides comparing to 0 has the advantage to be directly a boolean wheres your suggestion should actually look like `return (bool) count($this->payments->getPayments());`

Reply:
Be explicit is great but when you declare a return type in php 7, it will eventually be typecasting.
Your code look like this: `return (bool) ((bool) (count($this->payments->getPayments()) > 0));`